backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/payment/paymentallocation/service/CreditMemoInvoiceAsPaymentDocumentWrapper.java

```diff
@@ -111,16 +111,32 @@ public void addAllocatedAmt(final Money allocatedPayAmtToAdd)
 		creditMemoPayableDoc.addAllocatedAmounts(AllocationAmounts.ofPayAmt(allocatedPayAmtToAdd.negate()));
 	}
 
+	/**
+	 * Check only the payAmt as that's the only value we are allocating. see {@link  CreditMemoInvoiceAsPaymentDocumentWrapper#addAllocatedAmt(Money)}
+	 */
 	@Override
 	public boolean isFullyAllocated()
 	{
-		return creditMemoPayableDoc.isFullyAllocated();
+		return creditMemoPayableDoc.getAmountsToAllocate().getPayAmt().isZero();
 	}
 
+	/**
+	 * Computes projected over under amt taking into account discount.
+	 *
+	 * @implNote for credit memo as payment, the negated discount needs to be added to the open amount. Negated value is used
+	 * as it actually needs to increase the open amount.
+	 * 
+	 * e.g. Having a credit memo with totalGrandAmount = 10 and paymentTerm.Discount=10% translates to 11 total payment amount available.
+	 */
 	@Override
 	public Money calculateProjectedOverUnderAmt(@NonNull final Money payAmountToAllocate)
 	{
-		return creditMemoPayableDoc.computeProjectedOverUnderAmt(AllocationAmounts.ofPayAmt(payAmountToAllocate.negate()));
+		final Money openAmtWithDiscount = creditMemoPayableDoc.getOpenAmtInitial()
+				.add(creditMemoPayableDoc.getAmountsToAllocateInitial().getDiscountAmt().negate());
```

Comment:
imho adding the negated discount helps to better understand the logic as we are supposed to add the discount to the amount (but since we're talking about a credit memo the actual discount value needs to be negated). But, if you consider that it is easier to understand the logic by directly substracting the discount we can do it that way.

Reply:
For ease of understanding then please extract the discount amount to a local variable. This way you can keep the .add() logic and it doesn't hurt the eye as much.
Also please do the same below, where feasible.